### PR TITLE
Add debug info

### PR DESCRIPTION
**Pull request by hhugo**
The PR proposes to add debug info when compiling zarith (replicating default settings in dune)

### CODE REVIEW

**Comment by antoinemine** [comment]
I think that debug info should not enabled in the default target. Maybe we can add a debug target, or a ./configure option.

**Comment by hhugo** [comment]
Can you summarize the downside of enabling debug info ?

**Comment by xavierleroy** [comment]
> Unless you want to debug Zarith, you don't need debug info.

Not quite: if a Zarith function raises an exception, having debug info on Zarith produces a nicer backtrace for the exception.  That's a reason why most OCaml libraries are compiled with `-g` these days.

Note that in native-code, `-g` causes essentially zero performance degradation: optimizations are not turned off, but more static information is produced.  In bytecode, `-g` has a small performance impact, as a couple of optimizations are turned off.

**Comment by hhugo** [comment]
> Note that in native-code, -g causes essentially zero performance degradation: optimizations are not turned off, but more static information is produced. In bytecode, -g has a small performance impact, as a couple of optimizations are turned off.

That was my mental model.

@antoinemine, given xavier's answer, are you happy with the change ?

**Comment by antoinemine** [comment]
OK ! If this is OK for Xavier, we can merge.

**Comment by antoinemine** [comment]
Note that the -g flag seems to be propagated to gcc from ocamlmklib via ocamlc. However, the generated code for the C part of the library seems to be the same with and without -g flag, so, there is no obvious performance degradation (just bloated binaries and the fact that debug information on an optimized C binary may not be so useful for debugging). 

All the distributions I know (including source-based ones) strip binaries and provide debug info for C libraries in separate packages. I guess opam is different?
